WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt

```diff
@@ -0,0 +1,68 @@
+package org.wordpress.android.ui.domains
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestScope
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.verify
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_DOMAIN_TAPPED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_SHOWN
+import org.wordpress.android.ui.domains.management.DomainManagementViewModel
+import org.wordpress.android.ui.domains.management.DomainManagementViewModel.ActionEvent
+import org.wordpress.android.ui.domains.usecases.FetchAllDomainsUseCase
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+
+@ExperimentalCoroutinesApi
+class DomainManagementViewModelTest : BaseUnitTest() {
+    @Mock
+    private lateinit var analyticsTracker: AnalyticsTrackerWrapper
+
+    @Mock
+    lateinit var useCase: FetchAllDomainsUseCase
+
+    private lateinit var viewModel: DomainManagementViewModel
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+        viewModel = DomainManagementViewModel(testDispatcher(), analyticsTracker, useCase)
+    }
+
+    @Test
+    fun `WHEN ViewModel initialized THEN track DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_SHOWN event`() {
+        verify(analyticsTracker).track(DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_SHOWN)
+    }
+
+    @Test
+    fun `WHEN a domain is tapped THEN track DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_DOMAIN_TAPPED event`() {
+        viewModel.onDomainTapped(testDomain)
+        verify(analyticsTracker).track(DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_DOMAIN_TAPPED)
+    }
+
+    @Test
+    fun `WHEN a domain is tapped THEN send DomainTapped action event`() = testWithActionEvents { events ->
+        viewModel.onDomainTapped(testDomain)
+        advanceUntilIdle()
+        assertThat(events.last()).isEqualTo(DomainManagementViewModel.ActionEvent.DomainTapped(testDomain))
+    }
+
+    private fun testWithActionEvents(block: suspend TestScope.(events: List<ActionEvent>) -> Unit) =
+        test {
+            val actionEvents = mutableListOf<DomainManagementViewModel.ActionEvent>()
+            val job = launch { viewModel.actionEvents.toList(actionEvents) }
```

Comment:
I noticed that this same pattern is used in the purchase screen tests, and I've observed some failures with those ([on a PR which didn't have changes there](https://github.com/wordpress-mobile/WordPress-Android/pull/19431)). I'm not sure if this is the same reason the tests are failing on this PR 🤔 .

Do we need to pass the `testDispatcher` to `launch` here? Afaiu, the `block` (i.e. the unit test's body) will suspend until the job is cancelled, but I'm wondering whether `advanceUntilIdle()` will suspend long enough to allow the events list to become populated before the assertion? 🤔

Reply:
It seems this was not the reason after all. The missing mock (also missing on my PR 😅 ) were the culprit, but the real error was masked by the unmatched `when` exception.

Thanks for resolving this Antonis!